Sources/PostgresNIO/New/PostgresCodable.swift

```diff
@@ -3,30 +3,69 @@ import class Foundation.JSONEncoder
 
-    /// identifies the postgres format that is used to encode the value into `byteBuffer` in `encode`
+    /// Identifies the postgres format that is used to encode the value into `byteBuffer` in `encode`
     static var psqlFormat: PostgresFormat { get }
 
     /// Encode the entity into the `byteBuffer` in Postgres binary format, without setting
     /// the byte count. This method is called from the ``PostgresBindings``.
-    func encode<JSONEncoder: PostgresJSONEncoder>(into byteBuffer: inout ByteBuffer, context: PostgresEncodingContext<JSONEncoder>) throws
+    func encode<JSONEncoder: PostgresJSONEncoder>(
+        into byteBuffer: inout ByteBuffer,
+        context: PostgresEncodingContext<JSONEncoder>
+    ) throws
+}
+
+/// A type that can encode itself to a Postgres wire binary representation.
+/// Dynamic types are types that are not known at compile time. For example, a custom type that is created
+/// at runtime.
+public protocol PostgresDynamicTypeEncodable {
+    /// Identifies the data type that we will encode into `byteBuffer` in `encode`
+    ///
+    /// This is not static because the type is not known at compile time.
+    var psqlType: PostgresDataType { get }
+    /// identifies the postgres format that is used to encode the value into `byteBuffer` in `encode`
+    ///
+    /// This is not static because the type is not known at compile time.
+    var psqlFormat: PostgresFormat { get }
+
+    /// Encode the entity into the `byteBuffer` in Postgres binary format, without setting
+    /// the byte count. This method is called from the ``PostgresBindings``.
+    func encode<JSONEncoder: PostgresJSONEncoder>(
+        into byteBuffer: inout ByteBuffer,
+        context: PostgresEncodingContext<JSONEncoder>
+    ) throws
 }
 
 /// A type that can encode itself to a postgres wire binary representation. It enforces that the
 /// ``PostgresEncodable/encode(into:context:)-1jkcp`` does not throw. This allows users
 /// to create ``PostgresQuery``s using the `ExpressibleByStringInterpolation` without
 /// having to spell `try`.
-public protocol PostgresNonThrowingEncodable: PostgresEncodable {
+public protocol PostgresNonThrowingEncodable: PostgresEncodable, PostgresDynamicTypeNonThrowingEncodable {
     // TODO: Rename to `PostgresEncodable` with next major release
 
     func encode<JSONEncoder: PostgresJSONEncoder>(into byteBuffer: inout ByteBuffer, context: PostgresEncodingContext<JSONEncoder>)
 }
 
+/// A type that can encode itself to a Postgres wire binary representation.
+/// Dynamic types are types that are not known at compile time. For example, a custom type that is created
+/// at runtime.
+///
+/// This is the non-throwing alternative to ``PostgresDynamicTypeEncodable``. It allows users
+/// to create ``PostgresQuery``s using the `ExpressibleByStringInterpolation` without having to spell `try`.
+public protocol PostgresDynamicTypeNonThrowingEncodable: PostgresDynamicTypeEncodable {
```

Comment:
This should be `PostgresDynamicTypeEncodable`
```suggestion
public protocol PostgresDynamicTypeEncodable: PostgresDynamicTypeThrowingEncodable {
```